docs/signer.md

```diff
@@ -0,0 +1,354 @@
+# Kritis Signer
+
+Kritis Signer is a command-line tool that creates attestations. These attestations can be used for Kritis and Binary Authorization deployment enforcement.
+
+The tool can create attestations directly or conditionally, checking policy conformance based on Vulnerability scanning results from Google Cloud Container Analysis.
+
+It can be run either locally or as part of a continuous integration (CI) pipeline.
+
+Currently, the tool supports vulnerability-based policy check, 
+using [Google Vulnerability Scanning](https://cloud.google.com/container-registry/docs/vulnerability-scanning), a feature of  [Google Container Analysis](https://cloud.google.com/container-registry/docs/container-analysis) that returns a list of vulnerabilities associated with a container image. Kritis Signer then creates an attestation that is stored in the Google Container Analysis data store.
+Support for other type of checks (e.g., base-image check), other vulnerability sources and attestation storage are underway.
+
+This doc provides:
+- a general overview;
+- a step by step tutorial describing how to use the signer tool.
+
+## Overview
+
+### Signer Modes
+
+The tool can be run in three different modes:
+- **check-and-sign**: default mode, checks policy conformance for a image, and conditionally creates attestation for the image if check passes. Currently, only vulnerability policy is supported. 
+- **check-only**: only checks policy conformance and outputs results, but does not create any attestation.
+- **bypass-and-sign**: bypasses any check and creates attestation for a image.
+
+| mode            | checking? | signing? |
+|-----------------|-----------|----------|
+| check-and-sign  | ✓         | ✓        |
+| check-only      | ✓         |          |
+| bypass-and-sign |           | ✓        |
+
+Users can specify the mode with `-mode` flag.
+
+### Checking
+
+The signer tool can check an image's metadata against rules in a user-provided policy. The tool now supports vulnerability-based policy check
+and will add more checks in the future.
+
+#### Vulnerability Source
+
+During the checking process, the signer tool will fetch vulnerability results for the image from a source. 
+The tool now supports [Google Vulnerability Scanning](https://cloud.google.com/container-registry/docs/vulnerability-scanning) as a vulnerability source.
+If enabled, any image a user pushes to Google Container Registry, or gcr.io, will be automatically scanned for vulnerability. 
+
+#### Vulnerability Signing Policy
```

Comment:
We refer to this as a vulnerability policy in the docs

Reply:
We need to stick `Vulnerability Signing Policy` here as it's the spec in the source code.

Reply:
Ah, ok.  I'll need to update the doc.

Reply:
(Updated in the CGC doc as well.